### ISSUE TITLE
feat: put all files related to a VM in one place

### ISSUE THREAD

**Issue by blakemcbride** [wimpysworld/quickemu#1517]
**I confirm this feature has not been previously requested**
- [ x] I have searched the issues and this feature has not previously been requested

**Is your feature request related to a problem? Please describe.**

When you create a VM, all of the files related to that VM get put in a single directory.  This makes it easy to backup and move the VM.  The problem is that the `.conf` file gets placed one directory up.  This means one of the important files is in a different directory.  This muddies up the higher directory and makes it harder to move and backup the VM because you have to remember the `.conf` file.

**Describe the solution you'd like**

Move the `.conf` file into the same directory as its associated VM. When you start a VM, rather than specifying the `.conf` file, specify the directory. Quickemu can then find the .conf file in the directory (but don't depend on the directory name being the same as the file name; just look for a `.conf` file and assume there will only be one).

**Describe alternatives you've considered**

In addition to the `--vm` option now, you can add an option that will do what I am saying.  Thus you can be backward compatible.

**Additional context**

No additional context.

**Comment by philclifford** [comment]
There are many ways to not forget the .conf file when working with VMs.  This change would completely break `quickgui`.

**Comment by blakemcbride** [comment]
Yes, it is trivial to make a hard link from the config file into the VM directory.  `quickget` could easily do that.  The problem is, why dirty up the common directory with all of those `.conf` files?

It would be very easy to make it backward compatible.

I've never used `quickgui`.  `quickemu` has been fine for my needs. However, I am absolutely certain that `quickget`, `quickemu`, and `quickgui` could be trivially changes to accommodate my suggested architecture.  `quickemu` and `quickgui` could just use the new architecture and if it doesn't find the files it needs, simply look one directory up.

If there is interest, I'd be happy to learn `quickgui` and draw out the flow chart / algorithm.

**Comment by lj3954** [comment]
> There are many ways to not forget the .conf file when working with VMs. This change would completely break `quickgui`.

The solution I plan to implement for quickemu-rs is to have a 'vm_dir' field in the configuration, which will default to the filename (omitting extension), and accept either an absolute path or path relative to the config file's location (`.` would work for current directory). This would handle this request as well as allowing for more flexibility for other use cases, without resulting in a breaking change. I would explore the possibility of implementing the same in quickemu bash.

**Comment by shmahabi-dev** [comment]
This issue is being closed due to inactivity for 582 days. If this is still relevant, please comment or reopen and it will be re-opened. Thanks!

**Comment by shmahabi-dev** [comment]
This issue is being closed due to inactivity for 582 days. If this is still relevant, please comment or reopen and it will be re-opened. Thanks!

**Comment by bogiord** [comment]
I'd say this is still relevant, please don't close.
